src/ucp/tag/tag_send.c

```diff
@@ -148,7 +148,8 @@ ucp_tag_send_inline(ucp_ep_h ep, const void *buffer, size_t count,
 
     length = ucp_contig_dt_length(datatype, count);
 
-    if ((ssize_t)length <= ucp_ep_config(ep)->tag.max_eager_short) {
+    if ((ssize_t)length <= ucp_ep_config(ep)->tag.max_eager_short &&
+        ucp_memory_type_cache_is_empty(ep->worker->context)) {
```

Comment:
should we add a similar check to [rx check](https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_context.h#L332) ?